File: aspnet/GrabzIt/GrabzItException.cs
using System;
using GrabzIt.Enums;
using GrabzIt.COM;
using System.Runtime.InteropServices;

namespace GrabzIt
{
    [ClassInterface(ClassInterfaceType.None)]
    public class GrabzItException : Exception, IGrabzItException
    {
        public ErrorCode code;

        internal GrabzItException(string message, string code)
            : this(message, (ErrorCode)Convert.ToInt32(code))
        {
        }

        internal GrabzItException(string message, ErrorCode code)
            : base(message)
        {
            this.code = code;
        }

        internal GrabzItException(string message, ErrorCode code, Exception e)
            : base(message, e)
        {
            this.code = code;
        }

        public ErrorCode Code
        {
            get
            {
                return code;
            }
            set
            {
                Code = code;
            }
        }
    }
}

File: java/grabzit/src/main/java/it/grabz/grabzit/GrabzItException.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package it.grabz.grabzit;

import it.grabz.grabzit.enums.ErrorCode;

/**
 *
 * @author GrabzIt
 */
public class GrabzItException extends Exception {
    private ErrorCode code;

    public GrabzItException(String message, ErrorCode code, Exception innerException)
    {        
        super(message);
        this.initCause(innerException);
        this.code = code;
    }
    
    public GrabzItException(String message, ErrorCode code)
    {
        this(message, code, null);
    }
    
    public GrabzItException(String message, String code)
    {
        this(message, ErrorCode.values()[Integer.parseInt(code)]);
    }
    /**
     * @return the code
     */
    public ErrorCode getCode() {
        return code;
    }
    
    
}
